pom.xml

```diff
@@ -788,5 +807,48 @@ and
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>download-war</id>
+      <activation>
+        <file>
+          <missing>${env.JENKINS_WAR}</missing>
+        </file>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-dependency-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>jenkins-war</id>
+                <goals>
+                  <goal>copy</goal>
+                </goals>
+                <phase>process-test-resources</phase>
+                <configuration>
+                  <artifactItems>
+                    <artifactItem>
+                      <groupId>org.jenkins-ci.main</groupId>
+                      <artifactId>jenkins-war</artifactId>
+                      <version>${jenkins.version}</version>
+                      <type>war</type>
+                    </artifactItem>
+                  </artifactItems>
+                  <outputDirectory>${project.build.directory}</outputDirectory>
+                  <stripVersion>true</stripVersion>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
```

Comment:
If you are reading this PR from top to bottom, this hunk will only make sense after you have read the changes to `run.sh`.